File: Algorithmique appliquée/chapitre 1/exercice4.cpp
/*  programme sommePuissance(exercice4)
*   but: calcul de la somme de puissances de 2
*   date: 15/09/2016
*/
#include <stdio.h>
#include <iostream>
#include <math.h>
using namespace std;

int main(int argc, char *argv[]){

    unsigned int k,n, somme =0;

    cout << "entrez la valeur max= ";
    cin >> n;

    for(k=1;k <= n ;k++){
        somme += pow(2,k);
    }
    cout << "La somme des 2 puissances k = " << somme;
    return 0;
}

File: Algorithmique appliquée/chapitre1/python/exercice4.py
# -*-coding:utf-8 -*
# Programme exercice 4
# calcul de la somme de puissances de 2
# date : 15/09/2016

import os
import math


somme = 0

# saisie de la puissance maximale
n = int(input("entrez la valeur max = "))

# calcul et affichage de la somme des puissances de 2
for k in range(1, n+1) :
    somme += math.pow(2, k)
print("La somme des 2 puissance k = ", somme)


os.system("pause")
